### PR TITLE
ENH: support tpx 133 (GMX 2024.1)

### PR DESCRIPTION
**Pull request by tylerjereddy**
* One of my postdocs (Emvia) needs parsing of newer `.tpr` files to work; we actually are using tpx version `130` internally, but that wasn't from a stable release, so I just went for `2024.1` and presumbly we can just use `convert-tpr` to "up-version" to the stable release version internally.

* This patch used `convert-tpr` with GMX `2024.1` to mimic Jonathan's changes from gh-4052, when we last bumped `tpx` version. The full testsuite passed for me locally at least, so maybe we'll be lucky enough to have another simple bump.

* Note that I did not update the documentation or the `CHANGELOG`, fell free to push that in here if you're confident of the various tpx generation things, etc.

PR Checklist
------------
 - [X] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

## Developers certificate of origin
- [X] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4523.org.readthedocs.build/en/4523/

### CODE REVIEW

**Comment by IAlibay** [comment]
Thanks for working on this @tylerjereddy - I was also looking at this a few weeks back (got as far as John telling me to use convert-tpr 😅) and got distracted, I'll try to give it a review as soon as possible.

**Review by orbeckst — CHANGES_REQUESTED** [review]
If the test files are TPR 133 and all tests pass then that looks good enough for me.

We'd just need the docs + CHANGELOG. Could you quickly add those in, please?

**Comment by tylerjereddy** [comment]
> We'd just need the docs + CHANGELOG. Could you quickly add those in, please?

@orbeckst 

I pushed in what I could, but as I noted above, I don't know how to find out the `tpx generation` of a `.tpr` file (not the same as the `tpr version`). Comically, when I asked Google Gemini, it told me to use MDAnalysis to do it, presumably because it had slurped up this exact docs table. I just put in `??` for now--IMO it is the `.tpr` version proper that really means something to folks anyway.

**Comment by IAlibay** [comment]
@tylerjereddy if I remember correctly this is the enum class that gets used for the generation: https://gitlab.com/gromacs/gromacs/-/blob/main/src/gromacs/fileio/tpxio.cpp?ref_type=heads#L190

Should still be 28 I think? (sorry it's late and I'm not in a C++ frame of mind)

**Comment by IAlibay** [comment]
Just briefly putting this here because it's something @jbarnoud mentioned in a DM but I didn't get a chance to dig into it - apparently there are (edit: ITP?) input files for "all bonded" and "virtual sites" TPR tests somewhere that might need adding too?

**Comment by tylerjereddy** [comment]
> apparently there are input files for "all bonded" and "virtual sites" TPR tests somewhere that might need adding too?

pretty sure I did that in the diff already, I'll see if I can dig up the obscure generation thing

**Comment by IAlibay** [comment]
> > apparently there are input files for "all bonded" and "virtual sites" TPR tests somewhere that might need adding too?
> 
> pretty sure I did that in the diff already, I'll see if I can dig up the obscure generation thing

🤦🏾‍♂️ Sorry I'm not paying attention - I saw the dummy one but somehow my brain just ignored it completely

File: package/MDAnalysis/topology/TPRParser.py [hunk 1]
```diff
@@ -66,6 +66,7 @@
    122        28             2021                 yes
    127        28             2022                 yes
    129        28             2023                 yes
+   133        28             2024.1               yes
```
Comment (tylerjereddy) [inline package/MDAnalysis/topology/TPRParser.py:69]: Ok, I added the generation as `28` based on a fresh compilation of GROMACS `2024.1` with this adjustment:

```diff
--- a/src/gromacs/fileio/tpxio.cpp
+++ b/src/gromacs/fileio/tpxio.cpp
@@ -1034,6 +1034,7 @@ static void do_inputrec(gmx::ISerializer* serializer, t_inputrec* ir, int file_v
 
     ir->tpxFileVersion = file_version;
 
+    fprintf(stderr, "****** Note: tpx_generation %d\n", tpx_generation);
     if (file_version != tpx_version)
     {
         /* Give a warning about features that are not accessible */
```

Then used one of the `.tpr` files here as follows:

`gmx mdrun -s dummy_2024.tpr`

which includes in its output:

```
<snip>
Reading file dummy_2024.tpr, VERSION 2024.1-spack (single precision)
****** Note: tpx_generation 28
****** Note: tpx_generation 28
<snip>
```

**Review by IAlibay — APPROVED** [review]

**Review by orbeckst — APPROVED** [review]

**Pull request merged**